pkg/skaffold/runner/runner.go

```diff
@@ -206,7 +208,7 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*v1
 		return nil, errors.Wrap(err, "starting logger")
 	}
 
-	watcher := r.watchFactory(deployDeps, artifacts, PollInterval)
+	watcher := r.watchFactory(deployDeps, artifacts, PollInterval, r.opts)
```

Comment:
I'd love to see the `watchFactory` implementation begins different instead of passing opts here.
I tried to make sure opts don't need to be stored in the Runner by moving all the code related to opts in `NewRunner()`

Reply:
I agree --  I just did the lowest friction implementation to implement it without having too many other changes.  I'll change it to be done the right way